src/components/base/outline-modal/outline-modal.ts

```diff
@@ -0,0 +1,134 @@
+import { html, TemplateResult, CSSResultGroup } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import componentStyles from './outline-modal.css.lit';
+import { OutlineElement } from '../outline-element/outline-element';
+
+export type ModalSize = 'small' | 'medium' | 'full-screen';
+
+export const modalSizes: ModalSize[] = ['small', 'medium', 'full-screen'];
+
+/**
+ * The Outline Modal component
+ * @element outline-modal
+ * @slot default - The modal contents
+ * @slot outline-modal--trigger - The trigger for the modal
+ * @slot outline-modal--header - The header in the modal
+ */
+@customElement('outline-modal')
+export class OutlineModal extends OutlineElement {
+  static styles: CSSResultGroup = [componentStyles];
+
+  @property({ type: String })
+  size?: ModalSize = 'medium';
+
+  @property({ attribute: false })
+  isOpen = false;
+
+  @query('#overlay')
+  private overlayElement!: HTMLDivElement;
+
+  async open(): Promise<void> {
+    if (!this.isOpen) {
+      this.isOpen = true;
+
+      await this.updateComplete;
+
+      this.overlayElement.focus();
+
+      this.dispatchEvent(new CustomEvent('opened'));
+    }
+  }
+
+  async close(): Promise<void> {
+    if (this.isOpen) {
+      this.isOpen = false;
+
+      await this.updateComplete;
+
+      this.dispatchEvent(new CustomEvent('closed'));
+    }
+  }
+
+  private _handleModalTrigger(event: MouseEvent | KeyboardEvent): void {
+    let shouldOpen = false;
+
+    switch (event.type) {
+      case 'click':
+        shouldOpen = true;
+        break;
+      case 'keyup':
+        if ('key' in event && event.key === 'Enter') {
+          shouldOpen = true;
+          break;
+        }
+    }
+
+    if (shouldOpen) {
+      this.open();
+    }
+  }
+
+  private _handleModalClose(event: Event): void {
+    // Only trigger if we click directly on the event that wants to receive the click.
+    if (event.target === event.currentTarget) {
+      this.close();
+    }
+  }
+
+  private _handleOverlayKeyup(event: KeyboardEvent): void {
+    if (event.key === 'Escape') {
+      this.close();
+    }
+  }
+
+  private _overlayTemplate(): TemplateResult {
+    let template = html``;
+
+    if (this.isOpen) {
+      template = html`
+        <div
+          id="overlay"
+          tabindex="-1"
+          class="${this.size}"
+          @click="${this._handleModalClose}"
```

Comment:
Wondering if it might be worth it to make this conditional? In the case where you may want to force the user to take a specific action before closing the modal, rather than just being able to click the overlay to close it.

Reply:
It's a good question. I don't have strong opinions. Maybe we can discuss as a group?

Reply:
This would be a nice feature. Another prop of `canClose` or something, but this would require additional work too so we define what the confirmation button/link/text is/does/etc. 

If it's not "easy", I'd say we add this as a new feature request and iterate.

Reply:
Thanks for taking a look. https://designsystem.digital.gov/components/modal/ has an example of this we can use as a starting point. For now, I think making the close button optional and requiring the developer to configure the actions is easiest / most flexible and we could look at other options later?

Reply:
I've added this as an attribute `shouldForceAction`. There is a story for this in the latest version. Should remove the close features and adjusts the keyboard navigation for focus.